Source/Core/Core/Config/DefaultLocale.cpp

```diff
@@ -0,0 +1,201 @@
+// Copyright 2020 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Core/Config/DefaultLocale.h"
+
+#include <algorithm>
+#include <array>
+#include <locale>
+#include <optional>
+#include <string>
+#include <string_view>
+#include <vector>
+
+#ifdef _WIN32
+#include <Windows.h>
+#endif
+
+#include "Common/Assert.h"
+#include "Common/CommonTypes.h"
+#include "Common/StringUtil.h"
+#include "Core/Host.h"
+#include "DiscIO/Enums.h"
+
+namespace Config
+{
+static std::vector<std::string> GetPreferredLocales()
+{
+  static const std::vector<std::string> locales = Host_GetPreferredLocales();
+  return locales;
+}
+
+static std::optional<DiscIO::Language> TryParseLanguage(const std::string& locale)
+{
+  const std::vector<std::string> split_locale = SplitString(locale, '-');
```

Comment:
Are we sure we get something like `en-US` (with a dash) and not `en_US` (with an underscore) as `locale`?

Reply:
Based on Qt's documentation and the tests I've done, it seems like we're only getting `-`. But I admittedly haven't tested macOS or Linux.